plugins/cpp/parser/src/cppparser.cpp

```diff
@@ -491,7 +498,7 @@ void CppParser::initBuildActions()
   });
 }
 
-void CppParser::markAsModified(const model::FilePtr file_)
+void CppParser::markByInclusion(model::FilePtr file_)
```

Comment:
Well, that's an interesting point there. We assumed that all inclusion dependencies are contained by the `CppHeaderInclusion` table, since otherwise a couple of other existing functionalities (e.g. *inclusion graph*) would give an incorrect result.
In case we cannot take that for sure, we shall investigate this and I suggest that the `CppHeaderInclusion` table shall contain all inclusions, even the ones given by the `-include` flag.

Reply:
Okay, let's continue this discussion at the ticket #288 then. But currently without that information available it would be hard to fix this issue with incremental parsing.